test/widgets/store_test.dart

```diff
@@ -113,6 +119,45 @@ void main() {
       .contains('consider MaterialAccountWidgetRoute');
   });
 
+  testWidgets('PerAccountStoreWidget reset to choose-account page'
+              ' when logging out on invalid API key', (tester) async {
+    await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+    addTearDown(testBinding.reset);
+
+    const loadPerAccountDuration = Duration(seconds: 30);
+    assert(loadPerAccountDuration > kTryAnotherAccountWaitPeriod);
+    testBinding.globalStore.loadPerAccountDuration = loadPerAccountDuration;
+    testBinding.globalStore.loadPerAccountException = ZulipApiException(
+      routeName: '/register', code: 'INVALID_API_KEY', httpStatus: 400,
+      data: {}, message: '');
+    await tester.pumpWidget(const ZulipApp());
+    await tester.pump(); // start to load account
+    check(find.byType(CircularProgressIndicator)).findsOne();
+    check(find.byType(ChooseAccountPage)).findsNothing();
+    check(find.byType(BackButton)).findsNothing();
+
+    await tester.pump(kTryAnotherAccountWaitPeriod);
+    await tester.tap(find.text('Try another account'));
+    await tester.pump(); // tap the button
+    await tester.pump(const Duration(milliseconds: 250)); // wait for animation
+    check(find.byType(CircularProgressIndicator)).findsOne();
```

Comment:
I don't yet understand this `CircularProgressIndicator` check.
- Does the test expect it to be on a specific page? (meaning the check stops being as useful if another page adds a `CircularProgressIndicator`)
- If we're expecting it on a non-active/foregrounded page, do we need to use `skipOffstage: false`?

Reply:
It's looking for the route that is not on the top of the stack. I'm switching to a rewrite of this using `NavigatorObserver`'s, so that we can check the navigator stack more explicitly.